### PR TITLE
Add `key` methods to `SingleTableMetadata`

### LINKED ISSUE #880: Add key related methods to SingleTableMetadata
### Problem Description

_As a user, it would be useful to be able to programmatically set primary keys, sequence keys, sequences indices and alternate keys._

### Expected behavior

- Add `set_primary_key(id)` method
    - Parameters:
        - `id`: string of column name that is going to be the primary key, or tuple string column names
    - Should set the `primary_key` attribute in the dict
    - Warnings:
        - If there is already a key raise this warnings
        `Warning: There is an existing primary key 'user_id'. This key will be removed.`
```python
>>> metadata.set_primary_key(id='user_id')

# for a composite key, users can input a tuple
# Note: we chose to make the input a tuple bc it's immutable
>>> metadata.set_primary_key(id=('first_name', 'last_name'))
Warning: There is an existing primary key 'user_id'. This key will be removed.
```

 - Add `set_sequence_key(id)` method
     - Parameters:
        - `id`: string of column name that is going to be the sequence key
     - Updates the `sequence_key` in the dict
 - Add `set_sequence_index(column_name)` method
     - Parameters:
        - `column_name`: string of column name that is going to be the sequence index
     - Updates the `sequence_index` in the dict
- Add `add_alternate_keys(ids)` method
    - Parameters:
        - `ids`: A list of strings or tuples of string that represent column names
        - Updates the `alternate_keys` field in the dict
```python
>>> metadata.add_alternate_keys(ids=['username', 'email'])

# for alternate composite key, users can input tuples inside of lists
>>> metadata.add_alternate_keys(ids=[('first_name', 'last_name'), 'ssn'])
```

### Additional context

- For validation and errors, use the same validation as the `validate` method in #879.

### PR DESCRIPTION
**Pull request by fealho**
Resolve #880.

### CODE REVIEW

File: sdv/metadata/single_table.py [hunk 1]
```diff
@@ -242,6 +242,73 @@ def detect_from_csv(self, filepath, pandas_kwargs=None):
...
+
+    def set_sequence_key(self, id):
+        """Set the metadata sequence key.
+
+        Args:
+            id (str, tuple):
+                Name (or tuple of names) of the sequence key column(s).
+        """
+        is_tuple_of_str = isinstance(id, tuple) and all(isinstance(i, str) for i in id)
+        if not isinstance(id, str) and not is_tuple_of_str:
```
Comment (amontanez24) [inline sdv/metadata/single_table.py:289]: we can probably pull this into a method that they can all use that checks if an id is a str or tuple of strings

**Review by amontanez24 — APPROVED** [review]
One small comment but besides that it LGTM!

File: sdv/metadata/single_table.py [hunk 2]
```diff
@@ -118,8 +121,7 @@ def _validate_unexpected_kwargs(self, column_name, sdtype, **kwargs):
         expected_kwargs = self._EXPECTED_KWARGS.get(sdtype, ['pii'])
         unexpected_kwargs = set(list(kwargs)) - set(expected_kwargs)
         if unexpected_kwargs:
-            unexpected_kwargs = list(unexpected_kwargs)
-            unexpected_kwargs.sort()
+            unexpected_kwargs = sorted(unexpected_kwargs)
```
Comment (pvk-developer) [inline sdv/metadata/single_table.py:124]: 👍🏻

File: tests/unit/metadata/test_single_table.py [hunk 3]
```diff
@@ -787,6 +787,262 @@ def test_detect_from_csv_with_kwargs(self, mock_print):
...
+    def test_set_primary_key_tuple(self):
+        """Test that ``set_primary_key`` sets the ``_metadata['primary_key']`` value for tuples."""
+        # Setup
+        instance = SingleTableMetadata()
+
+        # Run
+        instance.set_primary_key(('column1', 'column2'))
+
+        # Assert
+        instance._metadata['primary_key'] == ('column1', 'column2')
```
Comment (pvk-developer) [inline tests/unit/metadata/test_single_table.py:899]: Missing `assert`

File: tests/unit/metadata/test_single_table.py [hunk 4]
```diff
@@ -787,6 +787,262 @@ def test_detect_from_csv_with_kwargs(self, mock_print):
...
+        instance = SingleTableMetadata()
+        instance._metadata['primary_key'] = 'column0'
+
+        # Run
+        instance.set_primary_key('column1')
+
+        # Assert
+        warning_msg = "There is an existing primary key 'column0'. This key will be removed."
+        warning_mock.warn.called_once_with(warning_msg)
+        instance._metadata['primary_key'] == 'column1'
```
Comment (pvk-developer) [inline tests/unit/metadata/test_single_table.py:924]: Missing `assert`

File: tests/unit/metadata/test_single_table.py [hunk 5]
```diff
@@ -787,6 +787,262 @@ def test_detect_from_csv_with_kwargs(self, mock_print):
...
+    def test_set_alternate_keys(self):
+        """Test that ``set_alternate_keys`` sets the ``_metadata['alternate_keys']`` value."""
+        # Setup
+        instance = SingleTableMetadata()
+
+        # Run
+        instance.set_alternate_keys(['column1', ('column2', 'column3')])
+
+        # Assert
+        instance._metadata['alternate_keys'] == ['column1', ('column2', 'column3')]
```
Comment (pvk-developer) [inline tests/unit/metadata/test_single_table.py:952]: Missing `assert`

File: tests/unit/metadata/test_single_table.py [hunk 6]
```diff
@@ -787,6 +787,262 @@ def test_detect_from_csv_with_kwargs(self, mock_print):
...
+    def test_set_sequence_key(self):
+        """Test that ``set_sequence_key`` sets the ``_metadata['sequence_key']`` value."""
+        # Setup
+        instance = SingleTableMetadata()
+
+        # Run
+        instance.set_sequence_key('column')
+
+        # Assert
+        instance._metadata['sequence_key'] == 'column'
```
Comment (pvk-developer) [inline tests/unit/metadata/test_single_table.py:980]: Missing `assert`

File: tests/unit/metadata/test_single_table.py [hunk 7]
```diff
@@ -787,6 +787,262 @@ def test_detect_from_csv_with_kwargs(self, mock_print):
...
+    def test_set_sequence_key_tuple(self):
+        """Test that ``set_sequence_key`` sets ``_metadata['sequence_key']`` for tuples."""
+        # Setup
+        instance = SingleTableMetadata()
+
+        # Run
+        instance.set_sequence_key(('column1', 'column2'))
+
+        # Assert
+        instance._metadata['sequence_key'] == ('column1', 'column2')
```
Comment (pvk-developer) [inline tests/unit/metadata/test_single_table.py:991]: Missing `assert`

File: tests/unit/metadata/test_single_table.py [hunk 8]
```diff
@@ -787,6 +787,262 @@ def test_detect_from_csv_with_kwargs(self, mock_print):
...
+        instance = SingleTableMetadata()
+        instance._metadata['sequence_key'] = 'column0'
+
+        # Run
+        instance.set_primary_key('column1')
+
+        # Assert
+        warning_msg = "There is an existing sequence key 'column0'. This key will be removed."
+        warning_mock.warn.called_once_with(warning_msg)
+        instance._metadata['sequence_key'] == 'column1'
```
Comment (pvk-developer) [inline tests/unit/metadata/test_single_table.py:1016]: Missing `assert`

File: tests/unit/metadata/test_single_table.py [hunk 9]
```diff
@@ -787,6 +787,262 @@ def test_detect_from_csv_with_kwargs(self, mock_print):
...
+    def test_set_sequence_index(self):
+        """Test that ``set_sequence_index`` sets the ``_metadata['sequence_index']`` value."""
+        # Setup
+        instance = SingleTableMetadata()
+
+        # Run
+        instance.set_sequence_index('column')
+
+        # Assert
+        instance._metadata['sequence_index'] == 'column'
```
Comment (pvk-developer) [inline tests/unit/metadata/test_single_table.py:1044]: Missing `assert`

File: tests/unit/metadata/test_single_table.py [hunk 10]
```diff
@@ -873,8 +1133,6 @@ def test__load_from_dict(self):
         # Assert
         assert instance._metadata == my_metadata
         assert instance._columns == {'my_column': 'value'}
-        assert instance._primary_key == 'pk'
-        assert instance._alternate_keys == []
```
Comment (pvk-developer) [inline tests/unit/metadata/test_single_table.py:877]: Can we add those back ?

File: tests/unit/metadata/test_single_table.py [hunk 11]
```diff
@@ -993,13 +1251,13 @@ def test_load_from_json(self, mock_json, mock_path, mock_open, mock_constraint):
...
             'alternate_keys': [],
+            'sequence_key': None,
+            'sequence_index': None,
             'constraints': [{'my_constraint': 'my_params'}],
             'SCHEMA_VERSION': 'SINGLE_TABLE_V1'
         }
         assert instance._columns == {'animals': {'type': 'categorical'}}
-        assert instance._primary_key == 'animals'
         assert instance._constraints == [{'my_constraint': 'my_params'}]
-        assert instance._alternate_keys == []
```
Comment (pvk-developer) [inline tests/unit/metadata/test_single_table.py:1002]: Can we add those back also ?

**Review by pvk-developer — APPROVED** [review]
Left a couple of nits in the tests but it looks good.

**Pull request merged**